decidim-core/lib/decidim/moderation_tools.rb

```diff
@@ -79,6 +79,24 @@ def hide!
       ) do
         @reportable.moderation.update!(hidden_at: Time.current)
         @reportable.try(:touch)
+
+        hide_comments!
+        send_notification_to_author if send_notification
```

Comment:
Those 2 methods needs to be outside the perform_action!. 
There is a transaction that happens there, and triggering them within a transaction may result in a concurency issue, when sidekiq or other backend job system would process the job faster that transaction commit ( I have seen it happening )